examples/v1beta1/pipelineruns/pipelinerun-with-pipelinespec-and-taskspec.yaml

```diff
@@ -6,6 +6,9 @@ spec:
   pipelineSpec:
     tasks:
       - name: echo-good-morning
+        metadata:
```

Comment:
this is its own field now? so it will not necessarily be written "just above" `taskSpec`. If we could have the `Task` as a field, optional metadata would work out of the box, as with `volumeClaimTemplate` where I can declare a `PersistentVolumeClaim` **with** or **without** metadata, https://github.com/tektoncd/pipeline/blob/master/pkg/apis/pipeline/v1beta1/workspace_types.go#L61
I would prefer that.

I also would like to be able to declare `PipelineRun` like that in `TriggerTemplate` ...so that I can set cutom labels, e.g. for commit-id.

Reply:
@jlpettersson I think your proposal would help avoiding adding `k8s` specific types in the `PipelineTask`, we can introduce a new field `TaskTemplate` or `EmbeddedTask` in `PipelineTask` which can of type `*Task`:

```
type PipelineTask struct {
	TaskTemplate *Task `json:"taskTemplate,omitempty"`
```

`PipelineTask` can have one of `TaskRef`, `TaskSpec`, or `TaskTemplate`.

I understand we are looking over to `Custom Tasks` which would help solve many use cases but this would solve the issue in short term and could be extended to custom tasks if needed. I am looking for feedback on this 🙏 

@vdemeester @bobcatfish @jlpettersson thoughts?

Reply:
I think something like that is the most proper solution, yes.
We haven't used "Template" for declarations in other places, but it may work. Other alternatives are:

`Task *Task `json:"task, omitempty"`

or

`TaskDeclaration *Task `json:"task, omitempty"`

Most correct description about what this is, is probably: _inline task declaration_

I don't think this is related to `CustomTask` at all, it will be the exact same problem. I think `CustomTask` recently has become the most misunderstood thing about Tekton.